internal/monitors/cpu/cpu_windows.go

```diff
@@ -0,0 +1,153 @@
+	maxBuffer := 2056
+	// buffer for results from the windows proc
+	resultBuffer := make([]systemProcessorPerformanceInformation, maxBuffer)
+	// size of the buffer in memory
+	bufferSize := uintptr(systemProcessorPerformanceInfoSize) * uintptr(maxBuffer)
+	// size of the returned response
+	var retSize uint32
+
+	// Invoke windows api proc.
+	// The returned err from the windows dll proc will always be non-nil even when successful.
+	// See https://godoc.org/golang.org/x/sys/windows#LazyProc.Call for more information
+	retCode, _, err := procNtQuerySystemInformation.Call(
+		systemProcessorPerformanceInformationClass, // System Information Class -> SystemProcessorPerformanceInformation
+		uintptr(unsafe.Pointer(&resultBuffer[0])),  // pointer to first element in result buffer
+		bufferSize,                        // size of the buffer in memory
+		uintptr(unsafe.Pointer(&retSize)), // pointer to the size of the returned results the windows proc will set this
+	)
+
+	// check return code for errors
+	if retCode != 0 {
+		return coreInfo, err
+	}
+
+	// calculate the number of returned elements based on the returned size
+	numReturnedElements := retSize / systemProcessorPerformanceInfoSize
+
+	// trim results to the number of returned elements
+	resultBuffer = resultBuffer[:numReturnedElements]
+
+	// convert all of the system processor information to gopsutil timesstats
+	coreInfo = make([]cpu.TimesStat, 0, numReturnedElements)
+	for core, info := range resultBuffer {
+		coreInfo = append(coreInfo, systemProcessorPerformanceInfoToCPUTimesStat(core, &info))
+	}
+
+	return coreInfo, nil
+}
+
+// getTimes has the same parameters and return values as gopsutil's cpu.Times() function
+// it actually relies on gopsutil's cpu.Times for non-per core information.
+// when percore is true it will attempt to use windows dll's to get the information
+// or fall back to gopsutil
+func getTimes(perCore bool) ([]cpu.TimesStat, error) {
+	// non-percore utilization in gopsutil does not rely on wmi so it's fine to
+	// utilize it as is
+	if !perCore {
+		return gopsutilTimes(perCore)
+	}
+	// Underneath the hood gopsutil relies on a wmi query for per core cpu utilization information
+	// this wmi query has proven to be problematic under unclear conditions.  It will hang
+	// from time to time, and when executed frequently.  Many projects rely on gopsutil for this information.
+	// Some have issues open complaining about hanging wmi calls, but none have a clear solution.
+	// In general if you search for information about WMI it is the best of a bad situation.  It
+	// is known to be buggy and slow.  A more performant solution is to
+	// get the System Processor Performance Information from the ntQuerySystemInformationfunction.
+
+	// attempt to use the windows api to get information on per core cpu times
+	res, err := ntQuerySystemInformation()
+	if err == nil {
+		return res, err
```

Comment:
This is misleading returning `err` when you know it is nil.  Just return `res, nil` and it is clearer.